itests/org.openhab.core.thing.tests/src/main/java/org/openhab/core/thing/firmware/Constants.java

```diff
@@ -131,19 +130,11 @@
             .withChangelog("English changelog valpha")
             .withOnlineChangelog(newURL("http://url/changelog-valpha?lang=en")).withModelRestricted(true).build();
 
-    private static URL newURL(String urlString) {
+    private static URL newURL(String url) {
         try {
-            return new URL(urlString);
+            return new URL(url);
         } catch (MalformedURLException e) {
-            throw new RuntimeException(e);
-        }
-    }
-
-    private static URI newURI(String urlString) {
-        try {
-            return new URI(urlString);
-        } catch (URISyntaxException e) {
-            throw new RuntimeException(e);
+            throw new IllegalArgumentException(e);
```

Comment:
I think `UncheckedIOException` is the better choice here.

Reply:
If it throws IllegalArgumentException you could also use URI.create(…) instead of catching URISyntaxException.

Reply:
OK I've switched it to an `UncheckedIOException`.

> If it throws IllegalArgumentException you could also use URI.create(…) instead of catching URISyntaxException and remove the whole method.

Unfortunately it's an URL and not an URI, so it doesn't have this method. You could create it as an URI and call `toURL()` but then again it will throw a `MalformedURLException`. :slightly_frowning_face: